pkg/image/registry/imagerepositorymapping/rest.go

```diff
@@ -73,17 +74,25 @@ func (s *REST) Create(ctx kapi.Context, obj runtime.Object) (runtime.Object, err
 	}
 
 	image := mapping.Image
-
-	if repo.Tags == nil {
-		repo.Tags = make(map[string]string)
+	tag := mapping.Tag
+	if len(tag) == 0 {
+		// TODO: redirect this to the stable tag
+		tag = "latest"
 	}
-	repo.Tags[mapping.Tag] = image.Name
 
 	if err := s.imageRegistry.CreateImage(ctx, &image); err != nil && !errors.IsAlreadyExists(err) {
 		return nil, err
 	}
-	if err := s.imageRepositoryRegistry.UpdateImageRepository(ctx, repo); err != nil {
-		return nil, err
+
+	next := api.TagEvent{
+		Created:              util.Now(),
+		DockerImageReference: image.DockerImageReference,
```

Comment:
You said that we'll eventually add a check to make sure that the user is allowed to access the image specified, correct?